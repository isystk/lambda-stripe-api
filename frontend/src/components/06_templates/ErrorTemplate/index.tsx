import HtmlSkeleton, {
  HtmlSkeletonProps,
  Title,
} from '@/components/05_layouts/HtmlSkeleton'
import { connect } from '@/components/hoc'
import React, { useState } from 'react'
import { ContainerProps } from 'types'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import { frontMenuItems } from '@/constants/menu'

/** ErrorTemplate Props */
export type ErrorTemplateProps = Omit<HtmlSkeletonProps, 'children'> & {
  statusCode: number
}
/** Presenter Props */
export type PresenterProps = ErrorTemplateProps

/** Presenter Component */
const ErrorTemplatePresenter: React.FC<PresenterProps> = ({
  statusCode,
  isMenuOpen,
  setMenuOpen,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>Page Not Found</Title>
    <Header
      isMenuOpen={isMenuOpen}
      setMenuOpen={setMenuOpen}
      menuItems={frontMenuItems}
    />
    <div className="flex items-center justify-center">
      <div className="p-3 md:p-32 w-full ">
        <div className="py-8">
          <p className="font-bold text-3xl md:mb-16 text-center">
            {statusCode} エラー
          </p>
          <div className="flex flex-wrap items-center p-8 md:mb-16">
            <div className="flex w-full justify-center items-center">
              <p className="text-2xl ">
                ページを正常に表示できませんでした。
                <br />
                ご入力内容等を確認の上で時間をおいて再度実行してください。
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div className="md:hidden">
      <SideMenu
        isMenuOpen={isMenuOpen}
        setMenuOpen={setMenuOpen}
        menuItems={frontMenuItems}
      />
    </div>
  </HtmlSkeleton>
)

/** Container Component */
const ErrorTemplateContainer: React.FC<
  ContainerProps<ErrorTemplateProps, PresenterProps>
> = ({ presenter, ...props }) => {
  const [isMenuOpen, setMenuOpen] = useState(false)
  return presenter({
    isMenuOpen,
    setMenuOpen,
    ...props,
  })
}

/** ErrorTemplate */
export default connect<ErrorTemplateProps, PresenterProps>(
  'ErrorTemplate',
  ErrorTemplatePresenter,
  ErrorTemplateContainer
)
