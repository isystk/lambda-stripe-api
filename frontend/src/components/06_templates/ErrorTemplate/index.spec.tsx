import React from 'react'
import renderer from 'react-test-renderer'
import ErrorTemplate, { ErrorTemplateProps } from './index'

describe('ErrorTemplate', () => {
  it('Match Snapshot', () => {
    const props: ErrorTemplateProps = { statusCode: 404 }
    const component = renderer.create(<ErrorTemplate {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
