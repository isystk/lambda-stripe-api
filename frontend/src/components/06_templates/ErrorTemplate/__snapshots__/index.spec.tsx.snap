// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTemplate Match Snapshot 1`] = `
[
  <header
    className="css-0 flex justify-between items-center h-16 px-4 py-4 sm:px-8 bg-main w-full"
  >
    <a
      className="css-0"
      href="http://localhost:9000/"
    >
      <div
        className="w-20 md:w-40"
      >
        <img
          className="css-s08p0c "
          src="/images/logo.png"
        />
      </div>
    </a>
    <div
      className="flex pr-3 z-50 md:hidden "
    >
      <button
        className="space-y-2 css-jgxxba"
        onClick={[Function]}
      >
        <div
          className="w-6 h-0.5 bg-black"
        />
        <div
          className="w-6 h-0.5 bg-black"
        />
        <div
          className="w-6 h-0.5 bg-black"
        />
      </button>
    </div>
    <nav
      className="hidden sm:block"
    >
      <ul
        className="flex gap-6"
      >
        <li>
          <a
            href="http://localhost:9000/"
            rel=""
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="http://localhost:9000/product/prod_NpvV9ohJIlgElI/payment"
            rel=""
          >
            Rates
          </a>
        </li>
        <li>
          <a
            href="https://blog.isystk.com/contact/"
            rel="noreferrer"
            target="_blank"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="flex items-center justify-center"
  >
    <div
      className="p-3 md:p-32 w-full "
    >
      <div
        className="py-8"
      >
        <p
          className="font-bold text-3xl md:mb-16 text-center"
        >
          404
           
          Error
        </p>
        <div
          className="flex flex-wrap items-center p-8 md:mb-16"
        >
          <div
            className="flex w-full justify-center items-center"
          >
            <p
              className="text-2xl "
            >
              The page could not be displayed properly. Please check the information you entered and try again in a few minutes.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="md:hidden"
  >
    <div
      className="css-0 fixed top-0 right-0 z-30 bg-base h-screen w-64 css-hc9ocb"
    >
      <div
        className="mt-24"
      />
      <ul
        className="list-none"
      >
        <li
          className="py-4"
        >
          <a
            className="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="http://localhost:9000/"
            rel=""
          >
            Home
          </a>
        </li>
        <li
          className="py-4"
        >
          <a
            className="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="http://localhost:9000/product/prod_NpvV9ohJIlgElI/payment"
            rel=""
          >
            Rates
          </a>
        </li>
        <li
          className="py-4"
        >
          <a
            className="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="https://blog.isystk.com/contact/"
            rel="noreferrer"
            target="_blank"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden z-20 bg-black opacity-50 fixed top-0 w-full h-screen"
      onClick={[Function]}
    />
  </div>,
]
`;
